### PR TITLE
Add basic database methods for user type

### PR DESCRIPTION
**Pull request by martijnjanssen**
Waiting on #13, should be rebased on master and merged there.

### CODE REVIEW

File: internal/pkg/database/queryer.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package database
+
+import (
+	"database/sql"
+)
+
+// Shared interface over both sqlx.Db and sqlx.Tx to enable
+// querying on both without considering the underlying type
+type Queryer interface {
```
Comment (skamoen) [inline internal/pkg/database/queryer.go:9]: Doesn't sqlx have something like `orm.DB`, like go-pg?

File: internal/pkg/database/user.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	if err != nil {
+		return dbAccessError(err)
+	}
+
+	return nil
+}
+
+func UserGetOnEmail(db Queryer, email string) (*User, error) {
+	user := &User{}
+	err := db.Get(user, "SELECT * FROM users WHERE email=$1", email)
```
Comment (skamoen) [inline internal/pkg/database/user.go:36]: This needs some sanitation I think

File: internal/pkg/database/queryer.go (hunk 1 above)
Comment (martijnjanssen) [reply internal/pkg/database/queryer.go:9]: Nope, this is the "go way"

File: internal/pkg/database/user.go (hunk 2 above)
Comment (martijnjanssen) [reply internal/pkg/database/user.go:36]: The $1 handles that :)

https://jmoiron.github.io/sqlx/#bindvars

**Review by skamoen — APPROVED** [review]

**Pull request merged**